src/structures/APIMessage.js

```diff
@@ -202,8 +198,7 @@ class APIMessage {
       content,
       tts,
       nonce,
-      embed: !isWebhookLike ? (this.options.embed === null ? null : embeds[0]) : undefined,
-      embeds: isWebhookLike ? embeds : undefined,
+      embeds: this.options.embeds ? embeds : undefined,
```

Comment:
```suggestion
      embeds: this.options.embeds?.map(embed => new MessageEmbed(embed).toJSON()),
```
With this, the additional logic above is also not needed anymore